torch_xla/csrc/python_util.cpp

```diff
@@ -0,0 +1,462 @@
+    TF_VLOG(1) << "GetClassNameAndObjFromFrame: Name exists - attribute not "
+                  "fetchable - clear error";
+    return;
+  }
+
+  auto c_name_str = py::str(c_name);
+  auto method_name_obj =
+      py::reinterpret_borrow<py::str>(frame->f_code->co_name);
+
+  if (c_name_str.ptr() == nullptr || method_name_obj.ptr() == nullptr) {
+    TF_VLOG(1) << "GetClassNameAndObjFromFrame: Name not a string";
+    return;
+  }
+
+  std::string likely_class_name = c_name.cast<std::string>();
+  std::string method_name = method_name_obj.cast<std::string>();
+
+  auto var_names =
+      py::reinterpret_borrow<py::object>(frame->f_code->co_varnames);
+  auto var_name_iterator = iter(var_names);
+  auto first_var_name = *var_name_iterator;
+
+  std::string var_0_name = "";
+
+  if (first_var_name.ptr() != nullptr) {
+    var_0_name = var_name_iterator->cast<std::string>();
+
+    if (var_0_name != "self") {
+      TF_VLOG(1)
+          << "GetClassNameAndObjFromFrame: Not a class method - var 0 name is "
+          << var_0_name << " (not 'self') and class name is "
+          << likely_class_name << " function/method name is " << method_name;
+      return;
+    }
+  }
+
+  TF_VLOG(1) << "GetClassNameAndObjFromFrame: Found class name = '"
+             << likely_class_name << "' for obs = " << AddrToString(obj)
+             << " function / method name = " << method_name;
+
+  sloc.class_name = likely_class_name;
+  sloc.obj_address =
+      (void*)obj.ptr();  // used to disambiguate objects - not dereferenced
+  TF_VLOG(1) << "GetClassNameAndObjFromFrame: Matched function name '"
+             << method_name << "' on class";
+
+  // Get the parent frame (if it exists) - we'll look for the object name here
+  PyFrameObject* parent_frame = nullptr;
+
+  // Reset global frame
+  parent_frame = frame->f_back;
+  int frame_id = 1;
+
+  while (!found_name && parent_frame != nullptr &&
+         parent_frame->f_code->co_argcount > 0 &&
+         parent_frame->f_localsplus[0] != nullptr) {
+    // Search object attributes
+    // e.g
+    // class A:
+    //   def __init__(self):
+    //     self.inner = B()
+    // Code returns 'inner'
+    TF_VLOG(1) << "GetClassNameAndObjFromFrame: Parent frame exists with at "
+                  "least one arg in frame="
+               << frame_id;
+
+    py::object parent_object =
+        py::reinterpret_borrow<py::object>(parent_frame->f_localsplus[0]);
+
+    // If the first arg in the parent frame is a class - then convert to
+    // dictionary and try and reverse lookup
+    found_name = ReverseSearchBreadthFirst(parent_object, obj, sloc.obj_name);
+
+    parent_frame = parent_frame->f_back;
+  }
+
+  // Check locals
+  PyFrame_FastToLocals(frame);
+
+  if (!found_name && frame->f_locals != nullptr) {
+    TF_VLOG(1) << "GetClassNameAndObjFromFrame: ** LOOK FOR LOCALS **";
+    py::object locals = py::reinterpret_borrow<py::object>(frame->f_locals);
+    if (!CheckNeuronInternal(frame))
```

Comment:
It's checking a hardcoded str now, can you make it generic

Reply:
Yeah - I was unsure how to proceed with what are basically "known infrastructure modules" - maybe a new python binding to set a global?